pipelines/build/common/build_base_file.groovy

```diff
@@ -397,95 +415,127 @@ class Builder implements Serializable {
-
-                // jdk11u-linux-x64-hotspot
-                def jobTopName = getJobName(configuration.key)
-                def jobFolder = getJobFolder()
-
-                // i.e jdk10u/job/jdk11u-linux-x64-hotspot
-                def downstreamJobName = "${jobFolder}/${jobTopName}"
-                context.echo "build name " + downstreamJobName
-
-                context.catchError {
-                    // Execute build job for configuration i.e jdk11u/job/jdk11u-linux-x64-hotspot
-                    context.stage(configuration.key) {
-                      context.echo "Created job " + downstreamJobName
-                      
-                      // execute build
-                      def downstreamJob = context.build job: downstreamJobName, propagate: false, parameters: config.toBuildParams()
-
-                      if (downstreamJob.getResult() == 'SUCCESS') {
-                          // copy artifacts from build
-                          context.node("master") {
-                              context.catchError {
-
-                                  //Remove the previous artifacts
-                                  context.sh "rm target/${config.TARGET_OS}/${config.ARCHITECTURE}/${config.VARIANT}/* || true"
-
-                                  context.copyArtifacts(
-                                          projectName: downstreamJobName,
-                                          selector: context.specific("${downstreamJob.getNumber()}"),
-                                          filter: 'workspace/target/*',
-                                          fingerprintArtifacts: true,
-                                          target: "target/${config.TARGET_OS}/${config.ARCHITECTURE}/${config.VARIANT}/",
-                                          flatten: true)
-
-                                  // Checksum
-                                  context.sh 'for file in $(ls target/*/*/*/*.tar.gz target/*/*/*/*.zip); do sha256sum "$file" > $file.sha256.txt ; done'
-
-                                  // Archive in Jenkins
-                                  context.archiveArtifacts artifacts: "target/${config.TARGET_OS}/${config.ARCHITECTURE}/${config.VARIANT}/*"
-                              }
-                          }
-                      } else if (propagateFailures) {
-                          context.error("Build failed due to downstream failure of ${downstreamJobName}")
-                          currentBuild.result = "FAILURE"
-                      }
+            def jobs = [:]
+
+            context.echo "Java: ${javaToBuild}"
+            context.echo "OS: ${targetConfigurations}"
+            context.echo "Enable tests: ${enableTests}"
+            context.echo "Enable Installers: ${enableInstallers}"
+            context.echo "Publish: ${publish}"
+            context.echo "Release: ${release}"
+            context.echo "Tag/Branch name: ${scmReference}"
+
+            jobConfigurations.each { configuration ->
```

Comment:
This is quite a complex construct now.  Consider extracting logic to local private functions that can be called from within this block

Reply:
Short of rebasing the entire function or passing in half of the params into these private functions, there's not much we can really do here to make it simpler. If we do decide to go ahead with that, I feel that should be done in a separate issue as that would be a big job.

Reply:
@johnoliver Ay insights here from your experience of pyramids of doom with Java?

Reply:
I think the only thing you can do is extract the functions and pass the mass of parameters, not much way around it